libs/wallet/src/web3modal/hooks/useIsMetaMask.ts

```diff
@@ -0,0 +1,7 @@
+import { useWeb3ModalProvider } from '@web3modal/ethers5/react'
+
+export function useIsMetaMask(): boolean {
+  const { walletProvider } = useWeb3ModalProvider()
+
+  return walletProvider?.isMetaMask ?? false
```

Comment:
Nitpick: Since we don't about `undefined`

```suggestion
  return !!walletProvider?.isMetaMask
```